ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -550,7 +550,13 @@ def create_data_reader(self, parent: Optional[Operator] = None, graph: Optional[
 
     @property
     def display_string(self):
-        return "Remote: " + self.url
+        return "URI: " + self.url
+
+    @property
+    def default_output_dir(self) -> Path:
+        if self.url.startswith("file://"):  # Might be // or ///
+            return Path(self.url.lstrip("file://").lstrip("/")).absolute().parent
```

Comment:
```suggestion
            return Path(self.url.removeprefix("file:").lstrip("/")).resolve().parent
```

`Path.resolve` also resolves symlinks.

Reply:
Would you recommend to generally replace all `absolute` usages with `resolve` or are there pitfalls that I'm overlooking?

Reply:
I think it's generally better to work with "canonical" paths if possible (no `.`/`..`, no symlinks, `/` instead of `\\` on Windows etc.) but maybe sometimes you have to keep `\\` because you are interfacing with other code, or you need to handle symlinks in a special way, so just doing `s/absolute/resolve/g` everywhere is probably not a good idea.

Reply:
By the way, is `lstrip("/")` really what do we want here? I've seen somewhere that `file:///my/file` (three slashes) gets mapped to absolute `/my/file`, but `file://my/file` (two slashes) gets mapped to relative `my/file`.

Reply:
I suppose overall while I thought this would be a neat feature, it introduces a lot of complexity because it's the first time we need to convert a URI to a `pathlib.Path`. It might be better to roll this back and let the default export folder be the project file's location.

Reply:
I'd not use `.resolve` for user paths/files. Usually there is a reason one symlinks to ones files.

Reply:
Stuck with `.absolute`, for above reason and because `.resolve` breaks some paths on Windows